libraries/AP_Notify/Display.cpp

```diff
@@ -533,16 +564,18 @@ void Display::update_ekf(uint8_t r)
 void Display::update_battery(uint8_t r)
 {
     char msg [DISPLAY_MESSAGE_SIZE];
-    AP_BattMonitor &battery = AP::battery();
+    AP_BattMonitor *battery = AP_BattMonitor::get_singleton();
+    if (battery == nullptr) {
```

Comment:
I recalled I'd written some code to fix this problem using defines, which the correct way to fix this.

Patch is here: https://github.com/ArduPilot/ardupilot/pull/26106/commits/a29575040f9053af66ea0ad35febb40967463fd4